cvat-ui/src/components/update-linked-cs-modal/select-cs-updating-scheme-modal.tsx

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useCallback, useState, useEffect } from 'react';
+import Modal from 'antd/lib/modal';
+
+import Button from 'antd/lib/button';
+
+import { Storage, getCore } from 'cvat-core-wrapper';
+
+import { useSelector, useDispatch } from 'react-redux';
+import { CombinedState } from 'reducers';
+import { updateProjectAsync, ProjectUpdateTypes } from 'actions/projects-actions';
+import { updateTaskAsync, TaskUpdateTypes } from 'actions/tasks-actions';
+import { useDropdownEditField } from 'utils/hooks';
+import CVATTooltip from 'components/common/cvat-tooltip';
+import Space from 'antd/lib/space';
+import { QuestionCircleOutlined } from '@ant-design/icons';
+import {
+    closeLinkedCloudStorageUpdatingModal as closeTaskLinkedCloudStorageUpdatingModal,
+} from '../../actions/tasks-actions';
+import { projectActions } from '../../actions/projects-actions';
+
+const core = getCore();
+
+function SelectCSUpdatingSchemeModal(): JSX.Element {
+    const task = useSelector((state: CombinedState) => state.tasks.updateWorkspace.instance);
+    const project = useSelector((state: CombinedState) => state.projects.updateWorkspace.instance);
+
+    const instance = task || project;
+    const [instanceType, setInstanceType] = useState('');
+    const dispatch = useDispatch();
+    const { stopEditField } = useDropdownEditField();
+
+    const saveInstance = useCallback(() => {
+        if (instance instanceof core.classes.Project) {
+            dispatch(updateProjectAsync(instance, ProjectUpdateTypes.UPDATE_ORGANIZATION));
+        } else if (instance instanceof core.classes.Task) {
+            dispatch(updateTaskAsync(instance, {}, TaskUpdateTypes.UPDATE_ORGANIZATION));
+        }
+    }, [instance]);
+
+    const closeModal = useCallback(() => {
+        if (instance instanceof core.classes.Project) {
```

Comment:
I’d suggest using `activities.updates` to store the instance and a simple boolean field to represent the open/closed state of this modal. Then, in this case, we can simply dispatch an action to set the boolean to `false`.

When we call `saveInstance` with `update*Async`, it will reset the corresponding `activities.updates` fields automatically.